### ISSUE TITLE
Feature request: Data filter

### ISSUE THREAD

**Issue by lp-dj** [opengisch/QField#2273]
I think an important feature that is still missing in QField is a filter function, similar to the one in QGIS.

Sure, you can predefine different filters in QGIS and store them in a separate map theme or in virtual layers and access them in QField this way.  However, this "workaround" falls short in some places and is often too inflexible.

What do you think about integrating a filter function in QField?

To keep it simple for the user and to avoid a previous configuration in QGIS, you could simply apply the filter to all matching layers after specifying the two parameters Filter column and Filter value.

I could imagine that for this purpose a control for the filter function could well be placed under the current theme selector:

Heading: Data filter

Parameter "Filter column": Here the user can select an attribute field. All unique attribute names contained in the project could appear here as a drop-down list. Preferably sorted alphabetically and with the possibility to type in an attribute via autocompletion, in case the project contains a lot of them and you would otherwise have to scroll for a long time.

Parameter "Filter value": Here the user can select an attribute value after which the records should be filtered. As a dropdown list, all unique values for the specified attribute could appear here. Preferably sorted alphabetically and with the possibility to type in a value via auto-completion, if the attribute contains a lot of values and you would otherwise have to scroll for a long time.

After specifying the two parameters, all layers that have a corresponding filter column will be filtered by the filter value. All layers that do not contain the filter column will not be filtered.

**Comment by Sekou225** [comment]
Its one of important fonction that miss in QField.

**Comment by cemno** [comment]
Yes, this would be really helpful, especially for large projects. An addition to the proposed enhancement would be the ability to add another row of filtering (combined with either "and" or "or") after filling in the first. 
This could be even further enhanced by adding a dropdown menu between the column and value selector where one can select "equal", "not equal", "larger than", "larger than or equal", "smaller than, "smaller than or equal", "contains", "doesn't contain", "is NULL", "is not NULL". 

Edit:
A good place to currently integrate that is the long tap action on the layer itself to activate a filter or set one in a separate window. Layers with an active filter should be clearly marked in the layer overview **and** the map canva aswell. Maybe a filter icon would be good which is also able to toggle an existing filter. If there is no existing filter, the icon shouldn't be present on the map canvas to not clutter the map.
